### DISCUSSION TITLE
前端导入的问题

### DISCUSSION THREAD

**Discussion by inkss** [ArtalkJS/Artalk#108] [Q&A] [answered]
利用文档的从 Twikoo 转换后的数据行囊大致长这个样子：

```json
{
    "id":{
        "$oid":"5bf40f137565710067989cf2"
    },
    "rid":"5bf23be844d9040061f9ca09",
    "content":"<a class=\"at\" href=\"#5bf23be844d9040061f9ca09\">@Anonymous </a><p>怎么个情况？👀</p>\n",
    "ua":"Mozilla/5.0 (X11; Linux x86_64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/70.0.3538.110 Safari/537.36",
    "is_collapsed":false,
    "created_at":"2018-11-20T13:41:39.872Z",
    "updated_at":"2020-04-16T08:33:10.625Z",
    "nick":"inkss",
    "email":"me@szyink.com",
    "link":"",
    "page_key":"/page/info/"
}
```

#### 两个问题

1.在前端导入时，如果填写目标站点，数据库 page.key 存的是 **站点内容 + page_key** 字段，希望前端页面能明确提示这一点。

2.page_key 内容为 `/page/info/`，实际导入的却为 `/page/info`。

**Comment by inkss** [comment]
数据库各种子外键关联，修改的时候真是太麻烦了。

**Reply by qwqcode** [reply]
外键这个下一个版本打算删了，没啥用，多类型数据库兼容性还不好

**Reply by inkss** [reply]
> 外键这个下一个版本打算删了，没啥用，多类型数据库兼容性还不好

确实，比如你现在评论是标记删除，数据库没删，然后导致页面表关联数据没法动了

**Reply by inkss** [reply]
另外，我现在导入的数据，pages表的title都是空的，后端能对比后更新下吗

**Reply by qwqcode** [reply]
> 另外，我现在导入的数据，pages表的title都是空的，后端能对比后更新下吗

ok，顺便加个 title 一键全部刷新功能

**Reply by qwqcode** [reply]
![image](https://user-images.githubusercontent.com/22412567/163796057-0076f244-9211-40bb-a7f6-1c7587981c16.png)

**Reply by qwqcode** [reply]
外键约束也删了，版本更新过后，可以把约束删了

或者直接 `artalk-go export ./artrans` 后，删除全部数据表，再 `artalk-go import ./artrans`，数据表会自动重建

**Comment by xiamuguizhi** [comment]
[子舒](https://imhan.cn/)[](https://qwqaq.com/)
2022-02-18
Chrome 98Windows 10
正在研究如何把https://example.com/1和https://example.com/1/指向同一个评论存储，目前测试发现，在有无斜杠的情况下，会产生两个不同的评论页面，这算一个大bug吧


[qwqcode](https://qwqaq.com/)[](https://imhan.cn/)
管理员
2022-02-18
Chrome 98Mac OS 10.15.7
传递的 PageKey 参数你可以前端处理一下，例如："https://example.com/1/".replace(/\/$/, "")

https://example.com/1 和 https://example.com/1/ 本来就不属于相同路径，后者等价于 https://example.com/1/index.html (和 web 服务器配置有关)

我认为这不算 bug


[子舒](https://imhan.cn/)
2022-02-18
Chrome 98Windows 10
为了不那么麻烦，我直接重定向

https://example.com/1 全部跳到 https://example.com/1/

非常好用！

不知道能否输入qq邮箱获取qq头像这个功能

![image](https://user-images.githubusercontent.com/19650297/163739897-b143290f-0c6e-4a1d-a63d-c7aaa91c77d4.png)



你看看有没有用，我那时候转过来没在意过这些...

**Reply by inkss** [reply]
我在数据库这边改了，不过把 `/1/` 导成 `/1` ，这个算是 bug 了。

![Inkss – pages  Tencent@SH _22-04-18_09-18-37](https://user-images.githubusercontent.com/31947043/163740060-3c7cd9a8-0ff6-4296-afc2-01c479ea4919.png)

**Reply by qwqcode** [reply]
确实。和 https://github.com/ArtalkJS/ArtalkGo/blob/master/lib/artransfer/importer.go#L180 有关联

**Comment by qwqcode** [comment, marked as answer]
已修复 https://github.com/ArtalkJS/ArtalkGo/commit/1e98599dc1513df59e428cc542599fb8cda61769
